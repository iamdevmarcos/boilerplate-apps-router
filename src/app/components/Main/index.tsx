const Main = () => (
  <main>
    <h1>Boilerplate Apps Router</h1>
  </main>
)

export default Main
