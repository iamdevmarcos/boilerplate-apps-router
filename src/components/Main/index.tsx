import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <h1>Boilerplate Apps Router</h1>
  </S.Wrapper>
)

export default Main
