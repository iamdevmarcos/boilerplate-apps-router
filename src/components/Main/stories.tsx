import { Meta, StoryObj } from '@storybook/react'

import Main, { type MainProps } from '.'

export default {
  title: 'components/Main',
  component: Main,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryObj<MainProps> = {}

export const Basic: StoryObj<MainProps> = {
  args: {
    title: 'title basic',
    description: 'description'
  }
}
