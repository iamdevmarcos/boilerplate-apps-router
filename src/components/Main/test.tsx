import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Main
        title={'boilerplate apps router'}
        description={'TypeScript, ReactJS, NextJS e Styled Components'}
      />
    )

    expect(
      screen.getByRole('heading', { name: /boilerplate apps router/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
