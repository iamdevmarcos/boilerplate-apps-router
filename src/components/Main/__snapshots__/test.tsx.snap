// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-bcPKhP jhrJfq"
>
  <h1>
    Boilerplate Apps Router
  </h1>
</main>
`;
